packages/expo-dev-menu/ios/ReactNativeCompatibles/ReactNative/DevClientAppDelegate.mm

```diff
@@ -28,12 +29,31 @@
 
 @interface RCTAppDelegate () <RCTComponentViewFactoryComponentProvider, RCTTurboModuleManagerDelegate>
 
-- (RCTRootViewFactory *)createRCTRootViewFactory;
-
 @end
 
 @implementation DevClientAppDelegate
 
+- (RCTRootViewFactory *)createRCTRootViewFactory
+{
+  RCTRootViewFactoryConfiguration *configuration =
+  [[RCTRootViewFactoryConfiguration alloc] initWithBundleURL:self.bundleURL
+                                              newArchEnabled:self.fabricEnabled
+                                          turboModuleEnabled:self.turboModuleEnabled
+                                           bridgelessEnabled:self.bridgelessEnabled];
+
+  __weak __typeof(self) weakSelf = self;
+  configuration.createRootViewWithBridge = ^UIView *(RCTBridge *bridge, NSString *moduleName, NSDictionary *initProps)
+  {
+    return [weakSelf createRootViewWithBridge:bridge moduleName:moduleName initProps:initProps];
+  };
+
+  configuration.createBridgeWithDelegate = ^RCTBridge *(id<RCTBridgeDelegate> self, NSDictionary *launchOptions)
+  {
+    return [weakSelf createBridgeWithDelegate:self launchOptions:launchOptions];
+  };
```

Comment:
```suggestion
  configuration.createBridgeWithDelegate = ^RCTBridge *(id<RCTBridgeDelegate> bridge, NSDictionary *launchOptions)
  {
    return [weakSelf createBridgeWithDelegate:bridge launchOptions:launchOptions];
  };
```
Let's rename it! It may be confusing to use `weakSelf` and `self` in the same line, and both are pointing to something else 😅

Reply:
fixed